File: DesignPatternsExercises/java/ProxyDesignPattern/Exercise/Movie.java
class Movie __________(1)__________ VideoFunctions {
    private String name;
    private int size;
    private String director;

    public Movie(String name) {
        this.name = name;
        this.size = 1023;
        this.director = "Christopher Nolan";
    }

    @Override
    public void playMovie() {
        System.out.println("Playing movie: " + name);
    }

    @Override
    public void stopMovie() {
        System.out.println("Stopping movie: " + name);
    }

    @Override
    public void deleteMovie() {
        System.out.println("Deleting movie: " + name);
    }

    @Override
    public String getMovieDetails() {
        return "Movie Details:\n    Name: " + name + "\n    Size: " + size + "MB\n    Director: " + director;
    }
}

File: DesignPatternsExercises/ruby/proxy_design_pattern/exercise/movie.rb
require_relative 'video_functions'

class Movie < VideoFunctions
  def __________(1)__________
    @name = name
    @size = 1023
    @director = 'Christopher Nolan'
  end

  def play_movie
    puts "Playing movie: #{@name}"
  end

  def stop_movie
    puts "Stopping movie: #{@name}"
  end

  def delete_movie
    puts "Deleting movie: #{@name}"
  end

  def movie_details
    "Movie Details:\n    Name: #{@name}\n    Size: #{@size} MB\n    Director: #{@director}"
  end
end
